agg-ast/schema_derivation/src/schema_derivation_tests/stage.rs

```diff
@@ -1141,3 +1141,243 @@ mod union {
         })
     );
 }
+
+mod replace {
+    use super::*;
+
+    test_derive_stage_schema!(
+        replace_root_nested_doc,
```

Comment:
Can you add a test where replace root is using a field which does not appear in the source schema to make sure things are working as expected.